### PR TITLE
backend/refactor: #147 Moved endride config to db

### PR DESCRIPTION
**Pull request by Dwibedi-10**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## Description
removed EndRideconfig from Dhall and fetching data from DB wherever required.

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [x] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs [hunk 1]
```diff
@@ -226,25 +223,22 @@ recalculateFareForDistance handle@ServiceHandle {..} booking ride recalcDistance
 
 getWaitingFare :: (MonadThrow m, Log m, MonadTime m, MonadGuid m) => ServiceHandle m -> Maybe UTCTime -> Maybe UTCTime -> Maybe Money -> m Money
 getWaitingFare ServiceHandle {..} mbTripStartTime mbDriverArrivalTime waitingChargePerMin = do
-  mbThresholdConfig <- findConfig
-  defaultThreshold <- getDefaultConfig <&> (.waitingTimeEstimatedThreshold)
-  let waitingTimeThreshold = fromMaybe defaultThreshold . join $ mbThresholdConfig <&> (.waitingTimeEstimatedThreshold)
+  thresholdConfig <- findConfig >>= fromMaybeM (InternalError "not found")
```
Comment (roman-bodavskiy) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs:226]: Could you add more specific error message?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Tabular/TransporterConfig.hs [hunk 2]
```diff
@@ -36,18 +36,18 @@ mkPersist
   [defaultQQ|
     TransporterConfigT sql=transporter_config
       merchantId MerchantTId
-      pickupLocThreshold Meters Maybe
-      dropLocThreshold Meters Maybe
+      pickupLocThreshold Meters
+      dropLocThreshold Meters
       rideTravelledDistThresholdWhenPickupOrDestIsDiff Meters Maybe
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Tabular/TransporterConfig.hs:41]: remove `rideTravelledDistThresholdWhenPickupOrDestIsDiff` and  `rideTravelledDistThresholdWhenPickupAndDestIsSame` if not used.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Tabular/TransporterConfig.hs [hunk 3]
```diff
@@ -36,18 +36,18 @@ mkPersist
   [defaultQQ|
     TransporterConfigT sql=transporter_config
       merchantId MerchantTId
-      pickupLocThreshold Meters Maybe
-      dropLocThreshold Meters Maybe
+      pickupLocThreshold Meters
```
Comment (roman-bodavskiy) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Tabular/TransporterConfig.hs:39]: you didn't add migration for these changes

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs [hunk 4]
```diff
@@ -268,12 +262,11 @@ getDistanceDiff booking distance = do
 
 isTimeOutsideOfThreshold :: (MonadThrow m, Log m, MonadTime m, MonadGuid m) => ServiceHandle m -> SRB.Booking -> DRide.Ride -> UTCTime -> m Bool
 isTimeOutsideOfThreshold ServiceHandle {..} booking ride now = do
-  mbThresholdConfig <- findConfig
+  thresholdConfig <- findConfig >>= fromMaybeM (InternalError "not found")
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs:265]: User proper error code that TransportConfigNotFound

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs [hunk 5]
```diff
@@ -226,25 +223,22 @@ recalculateFareForDistance handle@ServiceHandle {..} booking ride recalcDistance
...
+  let waitingTimeThreshold = thresholdConfig.waitingTimeEstimatedThreshold
       driverWaitingTime = fromMaybe 0 (diffUTCTime <$> mbTripStartTime <*> mbDriverArrivalTime)
       fareableWaitingTime = max 0 (driverWaitingTime / 60 - fromIntegral waitingTimeThreshold)
   pure $ roundToIntegral fareableWaitingTime * fromMaybe 0 waitingChargePerMin
 
 isPickupDropOutsideOfThreshold :: (MonadThrow m, Log m, MonadTime m, MonadGuid m) => ServiceHandle m -> SRB.Booking -> DRide.Ride -> LatLong -> m Bool
 isPickupDropOutsideOfThreshold ServiceHandle {..} booking ride tripEndPoint = do
   let mbTripStartLoc = ride.tripStartPos
-  mbThresholdConfig <- findConfig
+  thresholdConfig <- findConfig >>= fromMaybeM (InternalError "not found")
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs:235]: proper error code

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/dev/migrations/dynamic-offer-driver-app/0121-updated-transporter-config-table.sql [hunk 6]
```diff
@@ -0,0 +1,5 @@
+ALTER TABLE atlas_driver_offer_bpp.transporter_config DROP COLUMN ride_travelled_dist_threshold_when_pickup_or_dest_is_diff;
+ALTER TABLE atlas_driver_offer_bpp.transporter_config DROP COLUMN ride_travelled_dist_threshold_when_pickup_and_dest_is_same;
+UPDATE atlas_driver_offer_bpp.transporter_config SET pickup_loc_threshold = 500;
+UPDATE atlas_driver_offer_bpp.transporter_config SET drop_loc_threshold = 500;
+UPDATE atlas_driver_offer_bpp.transporter_config SET ride_time_estimated_threshold = 700;
```
Comment (hkmangla) [inline Backend/dev/migrations/dynamic-offer-driver-app/0121-updated-transporter-config-table.sql:5]: Add not null constraints as these values can't be maybe. Add not null on `waitingTimeEstimatedThreshold` too as you have removed maybe from it.

**Review by hkmangla — CHANGES_REQUESTED** [review]

**Review by hkmangla — APPROVED** [review]

**Pull request merged**